### PR TITLE
feat: improve tests [PHX-2768]

### PR DESCRIPTION
**Pull request by mayakarabula**
* Improve unit tests
* Add error handling that is dedicated to apply command

### CODE REVIEW

File: test/unit/commands/apply/index.test.ts [hunk 1]
```diff
@@ -42,10 +42,8 @@ describe('Apply Command', () => {
       cmd.catch(mockError)
     })
     .it('should inform that api keys need access to all compared environments', (ctx) => {
-      expect(ctx.stdout).to.contain('Changeset could not be created 💔')
-      expect(ctx.stdout).to.contain(
-        'An authorisation issue occurred. Please make sure the API key you provided has access to both environments.'
-      )
+      expect(ctx.stdout).to.contain('Merge was unsuccessful 💔')
+      expect(ctx.stdout).to.contain('An authorisation issue occurred. Please make sure the CMA token is correct.')
```
Comment (andreascful) [inline test/unit/commands/apply/index.test.ts:46]:
```suggestion
      expect(ctx.stdout).to.contain('An authorization issue occurred. Please make sure the CMA token is correct.')
```
I would suggest this variant is it is in line with other places we use this word.

**Review by andreascful — CHANGES_REQUESTED** [review]
Change looks good overall, just super small nit

File: test/unit/engine/apply-changeset/tasks/create-add-entities-task.test.ts [hunk 2]
```diff
@@ -39,7 +39,23 @@ describe('createAddEntitiesTask', () => {
       },
     }
 
-    context.client.cma.entries.create = sinon.stub().resolves(createdEntry)
+    class Spy {
```
Comment (andreascful) [inline test/unit/engine/apply-changeset/tasks/create-add-entities-task.test.ts:42]: Just curious why you chose to write it like this (with an ad-hoc class definition). Nothing wrong with that, just that I haven't seen that in our tests anywhere so far.

Comment (mayakarabula) [reply test/unit/engine/apply-changeset/tasks/create-add-entities-task.test.ts:42]: oh yeah, I was fighting a bit with sinon to make it work like both a spy and a stub and that was not really working out for some reason, so it was just faster to write a similar thing, I made it a class to be stateful and record number of calls

**Review by andreascful — APPROVED** [review]

**Pull request merged**